test/functional/system/rounds.js

```diff
@@ -0,0 +1,1691 @@
+				generatorPublicKey: Buffer.from(publicKey, 'hex'),
+			});
+			const account = _.find(accounts, {
+				publicKey: Buffer.from(publicKey, 'hex'),
+			});
+			if (!found) {
+				account.missedBlocks += 1;
+			}
+		});
+
+		return accounts;
+	}
+
+	function getExpectedRoundRewards(blocks) {
+		const rewards = {};
+
+		const feesTotal = _.reduce(
+			blocks,
+			(fees, block) => {
+				return new Bignum(fees).plus(block.totalFee);
+			},
+			0
+		);
+
+		const rewardsTotal = _.reduce(
+			blocks,
+			(reward, block) => {
+				return new Bignum(reward).plus(block.reward);
+			},
+			0
+		);
+
+		const feesPerDelegate = new Bignum(feesTotal.toPrecision(15))
+			.dividedBy(slots.delegates)
+			.floor();
+		const feesRemaining = new Bignum(feesTotal.toPrecision(15)).minus(
+			feesPerDelegate.times(slots.delegates)
+		);
+
+		__testContext.debug(
+			`	Total fees: ${feesTotal.toString()} Fees per delegates: ${feesPerDelegate.toString()} Remaining fees: ${feesRemaining} Total rewards: ${rewardsTotal}`
+		);
+
+		_.each(blocks, (block, index) => {
+			const publicKey = block.generatorPublicKey.toString('hex');
+			if (rewards[publicKey]) {
+				rewards[publicKey].fees = rewards[publicKey].fees.plus(feesPerDelegate);
+				rewards[publicKey].rewards = rewards[publicKey].rewards.plus(
+					block.reward
+				);
+			} else {
+				rewards[publicKey] = {
+					publicKey,
+					fees: new Bignum(feesPerDelegate),
+					rewards: new Bignum(block.reward),
+				};
+			}
+
+			if (index === blocks.length - 1) {
+				// Apply remaining fees to last delegate
+				rewards[publicKey].fees = rewards[publicKey].fees.plus(feesRemaining);
+			}
+		});
+
+		_.each(rewards, delegate => {
+			delegate.fees = delegate.fees.toString();
+			delegate.rewards = delegate.rewards.toString();
+		});
+
+		return rewards;
+	}
+
+	function addTransaction(transaction, cb) {
+		__testContext.debug(`	Add transaction ID: ${transaction.id}`);
+		// Add transaction to transactions pool - we use shortcut here to bypass transport module, but logic is the same
+		// See: modules.transport.__private.receiveTransaction
+		transaction = library.logic.transaction.objectNormalize(transaction);
+		library.balancesSequence.add(sequenceCb => {
+			library.modules.transactions.processUnconfirmedTransaction(
+				transaction,
+				true,
+				err => {
+					if (err) {
+						return setImmediate(sequenceCb, err.toString());
+					}
+					return setImmediate(sequenceCb, null, transaction.id);
+				}
+			);
+		}, cb);
+	}
+
+	function getNextForger(offset, cb) {
+		offset = !offset ? 1 : offset;
+		const lastBlock = library.modules.blocks.lastBlock.get();
+		const slot = slots.getSlotNumber(lastBlock.timestamp);
+		library.modules.delegates.generateDelegateList(
+			lastBlock.height + 1,
+			null,
+			(err, delegateList) => {
+				if (err) {
+					return cb(err);
+				}
+				const nextForger = delegateList[(slot + offset) % slots.delegates];
+				return cb(nextForger);
+			}
+		);
```

Comment:
There is a private method `getNextForger` in `./common.js` which is using `require(./common).getDelegateForSlot` for further logic. Please export `getNextForger ` function in `./common.js` and use it.